### PR TITLE
ROX-21973: compliance integration count endpoint

### PR DESCRIPTION
**Pull request by dashrews78**
## Description

Simply adds count endpoint to the compliance integration service

```
roxcurl /v2/compliance/integrations/count -u admin:SECRETSTUFF | json_pp
```

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

UT.  Also manually tested on a cluster.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: proto/api/v2/compliance_integration_service.proto [hunk 1]
```diff
@@ -27,11 +27,24 @@ message ListComplianceIntegrationsResponse {
...
     // ListComplianceIntegrations lists all the compliance operator metadata for the secured clusters
     rpc ListComplianceIntegrations (RawQuery) returns (ListComplianceIntegrationsResponse) {
         option (google.api.http) = {
             get: "/v2/compliance/integrations"
         };
     }
+
+    // GetComplianceIntegrationsCount returns the number of compliance operator integrations
+    // matching the given query
+    rpc GetComplianceIntegrationsCount(RawQuery) returns (CountComplianceIntegrationsResponse){
```
Comment (c-du) [inline proto/api/v2/compliance_integration_service.proto:45]: Is this count the number of clusters that has the integration?

**Review by bradr5 — APPROVED** [review]

File: central/complianceoperator/v2/integration/service/service_impl.go [hunk 2]
```diff
@@ -83,3 +84,20 @@ func (s *serviceImpl) ListComplianceIntegrations(ctx context.Context, req *v2.Ra
...
+// GetComplianceIntegrationsCount returns counts of profiles matching query
+func (s *serviceImpl) GetComplianceIntegrationsCount(ctx context.Context, request *v2.RawQuery) (*v2.CountComplianceIntegrationsResponse, error) {
+	parsedQuery, err := search.ParseQuery(request.GetQuery(), search.MatchAllIfEmpty())
+	if err != nil {
+		return nil, errors.Wrap(errox.InvalidArgs, err.Error())
+	}
+
+	integrationCount, err := s.complianceMetaDataStore.CountIntegrations(ctx, parsedQuery)
+	if err != nil {
+		return nil, errors.Wrap(errox.NotFound, err.Error())
```
Comment (c-du) [inline central/complianceoperator/v2/integration/service/service_impl.go:97]: nit: this error does not fit into "not found" category.

**Review by c-du — APPROVED** [review]

**Review by theencee — APPROVED** [review]

File: proto/api/v2/compliance_integration_service.proto (hunk 1 above)
Comment (dashrews78) [reply proto/api/v2/compliance_integration_service.proto:45]: essentially, yes.

**Pull request merged**